scripts/build_images.sh

```diff
@@ -15,11 +15,51 @@
 
 set -ex
 
-docker build -t php-nginx php-nginx
-docker build -t php56 testapps/php56
-docker build -t php56_custom testapps/php56_custom
-docker build -t php70 testapps/php70
-docker build -t php56_70 testapps/php56_70
-docker build -t php70_custom testapps/php70_custom
-docker build -t php56_nginx_conf testapps/php56_nginx_conf
-docker build -t php56_custom_configs testapps/php56_custom_configs
+TAG="${E2E_TEST_VERSION}"
+
+# Dump the credentials from the environment variable.
+php scripts/dump_credentials.php
+
+if [ -f "${PHP_DOCKER_GOOGLE_CREDENTIALS}" ]; then
+    # Use the service account for gcloud operations.
+    gcloud auth activate-service-account \
+        --key-file "${PHP_DOCKER_GOOGLE_CREDENTIALS}"
+    # Set the timeout
+    gcloud config set container/build_timeout 3600
+    SRC_TMP=$(mktemp -d)
+    BASE_IMAGE="gcr.io/${GOOGLE_PROJECT_ID}/php-nginx:${TAG}"
+fi
+
+build_image () {
+    if [ "$#" -ne 2 ]; then
+        echo "Two arguments; the image name and the dir are required"
+        exit 1
+    fi
+    IMAGE="${1}"
+    DIR="${2}"
+    if [ -f "${PHP_DOCKER_GOOGLE_CREDENTIALS}" ]; then
+        # Build the image with container builder service if we have
+        # credentials.
+        FULL_TAG="gcr.io/${GOOGLE_PROJECT_ID}/${IMAGE}:${TAG}"
+        SRC_DIR="${SRC_TMP}/${DIR}"
+        mkdir -p $(dirname ${SRC_DIR})
+        cp -R "${DIR}" "${SRC_DIR}"
+        # Replace the FROM line to point to our image in gcr.io.
+        sed -i -e "s|FROM php-nginx|FROM ${BASE_IMAGE}|" "${SRC_DIR}/Dockerfile"
+        gcloud -q alpha container builds create "${SRC_DIR}" --tag "${FULL_TAG}"
```

Comment:
It's not strictly needed, but if you run this locally, it will rewrite the source code.